bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -278,8 +278,8 @@ where
 						xcm: vec![
 							// Buy execution on target.
 							BuyExecution { fees: dest_para_fee_asset, weight_limit: Unlimited },
-							// Deposit asset to beneficiary.
-							DepositAsset { assets: Definite(asset.into()), beneficiary },
+							// Deposit both asset and fees to beneficiary.
+							DepositAsset { assets: Wild(All), beneficiary },
```

Comment:
This introduces a new failure mode and extra burden on integrations. The failure it is trying to fix is that it will create the account on the destination parachain. However this now places a burden on the destination parachain to store excess DOT in one of their pallets, if it is not configured to than the deposit will fail and assets will be trapped.

So we probably shouldn't do this on the destination chain portion. Only on asset hub.

Reply:
Its also hard to review, without seeing the complete messages that are forwarded to AssetHub and the final dest para,

So the RFC doc should include pseudocode describing the full messages.

Reply:
Yeah, that make sense. 

Usually I check the xcm message from log of emulated test with log level specified(e.g. RUST_LOG=xcm=trace) but it's not easy for others to review/understand.

Sorry for that, let me add the document with the xcm messages forwarded.

Reply:
https://github.com/Snowfork/polkadot-sdk/pull/137/commits/7e0ddaa4fbd8adbb61ea02a0e2f9b1bab8445a65 for the fix.

Reply:
https://github.com/Snowfork/snowbridge/pull/1174/commits/02ab816ba99bc009362b1836ba828a20826ae2c1 for the RFC

Reply:
Btw, the initial intention is just for deposit the excess DOT to beneficiary on destination chain, because it's always not good to leave the fees trapped somewhere instead of send it to the user. 

Maybe it seems much better to encourage the end user to send enough `destination_fee` to ensure the xcm will be success,  and we just let them know the fees will be finally refunded and no worry.